src/Client/Adapter/Curl.php

```diff
@@ -209,6 +209,13 @@ public function connect($host, $port = 80, $secure = false)
             }
         }
 
+        if (isset($this->config['sslcafile']) && $this->config['sslcafile']) {
```

Comment:
CURLOPT_SSL_VERIFYPEER is true by default, so if `sslverifypeer` is not set it's not a problem for CA. The issue comes if `sslverifypeer` is set to false and we set CA. In this case I think we should throw an Exception, instead of a silent action as you proposed. This is a clear misuse of the component,with potential security implications. 
@mlocati why you are against using an Exception for this case?